### PR TITLE
[OFFAPPS-802] Refresh App on parent ticket status change

### PR DESCRIPTION
**Pull request by vaevictis**
# Refresh App on parent ticket status change
## Jira ticket
[This way please](https://zendesk.atlassian.net/browse/OFFAPPS-802)
## Description
If the parent ticket got closed while "stay on ticket" was selected, the linked ticket app would still allow creation of a linked ticket, at least until the app / full page was reloaded.

This PR redirects the `ticket.status.changed` app event to a new function that fires a re-initialisation of the app if the new ticket status is `closed`.

### CODE REVIEW

File: spec/src/app_spec.js [hunk 1]
```diff
@@ -57,9 +57,51 @@ describe('App', () => {
     })
   })
 
+  describe('#onActivated', () => {
+    it('fetches main ticket data and sets locally')
```
Comment (ocke) [inline spec/src/app_spec.js]: what does this test exactly?

Comment (vaevictis) [reply spec/src/app_spec.js]: They are pending specs. I'm going to remove them from this PR as they are not directly related. 

If you'd scroll down this file,you'd see plenty of pending specs like this one. When the test runner, well, _runs_, it shows them this way:
pending specs
- I find it useful to write down as many test scenarios as possible, even without implementing them. The sheer act of writing them helps me ensure I've covered more edge cases than by just writing some code and hoping for the best. That was one of an early retro actions I took: When not knowing what an app does, writing scenarios first helps me learn about it.

- Also, I've effectively been filling the scenarios as I was adding features / fixing bugs in this app.

- Finally I reckon they are a descent documentation of the functions they are attached to. With the usual caveats of documentation. They need to be updated / deleted / implemented at some point.

Comment (ocke) [reply spec/src/app_spec.js]: maybe i should start doing the same. good idea.
